src/platform/packages/shared/kbn-unified-tabs/src/utils/calculate_responsive_tabs.ts

```diff
@@ -26,16 +26,22 @@ export const calculateResponsiveTabs = ({
   const availableContainerWidth =
     (containerWidth || window.innerWidth) - (hasReachedMaxItemsCount ? 0 : PLUS_BUTTON_SPACE);
 
-  let calculatedTabWidth =
-    items.length > 0 ? availableContainerWidth / items.length : MAX_TAB_WIDTH;
+  let wasSpaceEquallyDivided = items.length > 0;
+  let calculatedTabWidth = wasSpaceEquallyDivided
+    ? availableContainerWidth / items.length
+    : MAX_TAB_WIDTH;
 
   if (calculatedTabWidth > MAX_TAB_WIDTH) {
     calculatedTabWidth = MAX_TAB_WIDTH;
+    wasSpaceEquallyDivided = false;
   } else if (calculatedTabWidth < MIN_TAB_WIDTH) {
     calculatedTabWidth = MIN_TAB_WIDTH;
+    wasSpaceEquallyDivided = false;
   }
 
-  const numberOfVisibleItems = Math.floor(availableContainerWidth / calculatedTabWidth);
+  const numberOfVisibleItems = wasSpaceEquallyDivided
+    ? items.length
+    : Math.floor(availableContainerWidth / calculatedTabWidth);
```

Comment:
For example for 7 open tabs and browser window width around 1374px, number of possible visible items was calculated as `6.999999999999999` and rounded to 6 here while it should be all 7.